### ISSUE TITLE
Segfault in kafka-controller on release-1.13 when reconciling consumergroup

### ISSUE THREAD

**Issue by Cali0707** [knative-extensions/eventing-kafka-broker#3850]
**Describe the bug**
When running upgrade tests locally, I came across the following error:
```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1b9cd1e]

goroutine 474 [running]:
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup.(*Reconciler).schedule(0xc0006442d0, {0x2505580, 0xc001d96330}, 0xc00158e000)
	knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup/consumergroup.go:417 +0x1fe
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup.(*Reconciler).FinalizeKind(0xc0006442d0, {0x2505580, 0xc001d96330}, 0xc00158e000)
	knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup/consumergroup.go:232 +0x90
knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/injection/reconciler/eventing/v1alpha1/consumergroup.(*reconcilerImpl).Reconcile(0xc0005c23c0, {0x2505580, 0xc001d96300}, {0xc0010b9200, 0x57})
	knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/injection/reconciler/eventing/v1alpha1/consumergroup/reconciler.go:249 +0x3b8
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc0005da900)
	knative.dev/pkg@v0.0.0-20231103063133-e287426d1833/controller/controller.go:542 +0x4ad
knative.dev/pkg/controller.(*Impl).RunContext.func3()
	knative.dev/pkg@v0.0.0-20231103063133-e287426d1833/controller/controller.go:491 +0x59
created by knative.dev/pkg/controller.(*Impl).RunContext in goroutine 510
	knative.dev/pkg@v0.0.0-20231103063133-e287426d1833/controller/controller.go:489 +0x349
```

 **Expected behavior**
No panic should occur

 **To Reproduce**
Steps to reproduce the behavior.

 **Knative release version**

 **Additional context**
Add any other context about the problem here such as proposed priority
[kafka-controller-panic.txt](https://github.com/knative-extensions/eventing-kafka-broker/files/15067743/kafka-controller-panic.txt)

**Comment by Cali0707** [comment]
/cc @pierDipi 

I'm not sure what's causing this one :(

**Comment by Cali0707** [comment]
Ah, I had not fetched the latest release tag on my local branch (I had 1.13.7, not 1.13.8)...

This should be resolved on 1.13.8, so closing this
